src/migrations/00012-add-device-types.sql

```diff
@@ -0,0 +1,302 @@
+-- create device types table
+
+CREATE TABLE IF NOT EXISTS "device type" (
+	"created at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+,	"modified at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+,	"id" SERIAL NOT NULL PRIMARY KEY
+,	"slug" VARCHAR(255) NOT NULL UNIQUE
+,	"name" VARCHAR(255) NOT NULL
+);
+
+-- populate device types table
+
+INSERT INTO "device type" ("slug", "name")
+	SELECT "dt" as "slug", "dt" as "name" FROM
+	(
+		select dt FROM
+		(
+			SELECT lower("device type") as "dt" FROM "device"
+			UNION
+			SELECT lower("device type") as "dt" FROM "application"
+		) as DT_TABLE
+	) as FULL_DT;
+
+
+-- insert current dt from s3 as dts
+INSERT INTO "device type" ("slug", "name") VALUES
+  (
+    'am571x-evm',
+    'AM571X EVM (DISCONTINUED)'
+  ),
+  (
+    'apalis-imx6q',
+    'Apalis iMX6q (DISCONTINUED)'
+  ),
+  (
+    'artik533s',
+    'Samsung Artik 530s 1G (EXPERIMENTAL)'
+  ),
+  (
+    'odroid-xu4',
+    'ODROID-XU4'
+  ),
+  (
+    'asus-tinker-board',
+    'Asus Tinker Board (EXPERIMENTAL)'
+  ),
+  (
+    'blackboard-tx2',
+    'Nvidia blackboard TX2'
+  ),
+  (
+    'imx7-var-som',
+    'Variscite VAR-SOM-MX7 (EXPERIMENTAL)'
+  ),
+  (
+    'colibri-imx6dl',
+    'Colibri iMX6dl (DISCONTINUED)'
+  ),
+  (
+    'imx6ul-var-dart',
+    'Variscite DART-6UL (EXPERIMENTAL)'
+  ),
+  (
+    'jetson-tx1',
+    'Nvidia Jetson TX1 (EXPERIMENTAL)'
+  ),
+  (
+    'ccon-01',
+    'CloudConnector 01 (EXPERIMENTAL)'
+  ),
+  (
+    'orangepi-plus2',
+    'Orange Pi Plus2 (EXPERIMENTAL)'
+  ),
+  (
+    'raspberry-pi',
+    'Raspberry Pi (v1 and Zero)'
+  ),
+  (
+    'fincm3',
+    'Balena Fin (CM3)'
+  ),
+  (
+    'qemux86-64',
+    'QEMU X86 64bit (EXPERIMENTAL)'
+  ),
+  (
+    'orange-pi-zero',
+    'Orange Pi Zero (EXPERIMENTAL)'
+  ),
+  (
+    'kitra520',
+    'RushUp Kitra 520 (DISCONTINUED)'
+  ),
+  (
+    'jetson-tx2',
+    'Nvidia Jetson TX2'
+  ),
+  (
+    'artik10',
+    'Samsung Artik 10 (DISCONTINUED)'
+  ),
+  (
+    'qemux86',
+    'QEMU X86 32bit (EXPERIMENTAL)'
+  ),
+  (
+    'parallella',
+    'Parallella (DISCONTINUED)'
```

Comment:
I would expect us not adding discontinued device types in the DB. Even in case that a migrating user still uses them, we do insert the `FULL_DT` above based on all their old `device_type` records.
That's definitely not a blocker for me though, if we think adding them anyway will be safer.

Reply:
If we don't include it in the migration, they will be synced through s3 during runtime.

Reply:
Not only that, but the migration will fail as well, because there won't be a device-type row matching the app's/device's device type slug.

Reply:
@dfunckt there will be a match, since right above we do a union+toLowerCase with all current device type slugs in the DB.
@afitzek fair enough :+1: